File: cpp/NetworkDevice.cpp
// NetworkDevice.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

class NetworkDevice {
protected:
    string networkProtocol;
    float speedMbps;
    
public:
    NetworkDevice() {}
    
    NetworkDevice(string networkProtocol, float speedMbps) {
        this->networkProtocol = networkProtocol;
        this->speedMbps = speedMbps;
    }
    
    void setNetworkProtocol(string networkProtocol) {
        this->networkProtocol = networkProtocol;
    }
    
    void setSpeedMbps(float speedMbps) {
        this->speedMbps = speedMbps;
    }
    
    string getNetworkProtocol() {
        return this->networkProtocol;
    }
    
    float getSpeedMbps() {
        return this->speedMbps;
    }
    
    ~NetworkDevice() {}
};

File: python/NetworkDevice.py
class NetworkDevice() :
    def __init__(self, networkProtocol, speedMbps):
        
        self.__networkProtocol = networkProtocol
        self.__speedMbps = speedMbps
        
    def get_networkProtocol(self):
        return self.__networkProtocol
    def get_speedMbps(self):
        return self.__speedMbps
    
    # Setter methods
    def set_networkProtocol(self, networkProtocol):
        self.__networkProtocol = networkProtocol
    def set_speedMbps(self, speedMbps):
        self.__speedMbps = speedMbps